File: mobile/lib/class/sound.dart
import 'package:mobile/utilities/network.dart';

/// @class Sound
///
/// @brief Gère les données d'une musique
///
/// @file sound.dart
class Sound {

  late String id;
  late String title;
  late String artist;
  late int type;
  late String image;
  late String link;

  Sound(String id, String title, String artist, int type, String image, String link){
    this.id = id;
    this.title = title;
    this.artist = artist;
    this.type = type;
    if(image.startsWith("http://localhost")) {
      this.image = image.replaceAll("localhost", Network.hostName);
    } else {
      this.image = image;
    }
    if(link.startsWith("http://localhost")) {
      this.link = link.replaceAll("localhost", Network.hostName);
    } else {
      this.link = link;
    }
  }
}

File: tests/api/class/sound.js
class Sound {
	constructor(id, title, artist, type, image, link) {
		this.id = id;
		this.title = title;
		this.artist = artist;
		this.type = type;
		this.image = image;
		this.link = link;
	}

	isValid() {
		if (!this.id || typeof this.id !== 'string') return false;
		if (!this.title || typeof this.title !== 'string') return false;
		if (!this.artist || typeof this.artist !== 'string') return false;
		if ((parseInt(this.type) != 0 && parseInt(this.type) != 1) || typeof this.type !== 'string') return false;
		if (!this.image || typeof this.image !== 'string') return false;
		if (!this.link || typeof this.link !== 'string') return false;

		return true;
	}

	static toClass(data) {
		return new Sound(data['id'], data['title'], data['artist'], data['type'], data['image'], data['link']);
	}
}

module.exports = { Sound };
